### DISCUSSION TITLE
How to add new schemas as exposed schemas programmatically?

### DISCUSSION THREAD

**Discussion by pidanou** [supabase/supabase#15956] [Questions]
Hi guys,

I am building a multi-tenant app, where a tenant can dynamically ask to create a schema with the name he desires. My app creates the schema after some validation, automatically seeds the schema, add the required policies etc...

Of course I have a front-end that needs to access data on those schema. As far as I understand, to be able to use the Supabase API, the schema needs to exposed, and to do so I need to go the API settings on the console.

Is there a way to add the created schema without doing that manual action?

I have looked for the answer, but didn't see any answer that was working, for example using the Postgrest configuration, but in my case I use Supabase Cloud so I don't think I have access to the configuration and environment variables to configure Postgrest as I desire.

Thanks a lot!

**Comment by GaryAustin1** [comment]
The only way is going to be if the PostgREST schema setting can be changed in SQL.
https://postgrest.org/en/stable/references/api/schemas.html?highlight=schema#schemas
And this change to allow it to be changed for cloud environments.   (requires PostgREST 11)
https://postgrest.org/en/stable/references/configuration.html#in-database-configuration
In reading it seems like Supabase has to have setup a config to call the function described in the 2nd link.  I do not know if they have and have not tested it.

**Reply by pidanou** [reply]
I believe there are many ways of configuring Postgrest but I'm not sure it is in-database, I didn't find any trace of that table or any function that has somewhat a name that could be doing that (I searched using Pgadmin on the database).

**Reply by GaryAustin1** [reply]
Yeah the 2nd way is in database, it was just added.  The example shows setting the schema on the fly in the function.
BUT I don't think Supabase is setup to call the function or we still don't have permissions to do it.  It takes a setting to call the function from the looks of it.

**Reply by pidanou** [reply]
Ok I saw there was a enhancement proposal here that is exactly what I need: https://github.com/orgs/supabase/discussions/12270
